app/utils/router/helpers.ts

```diff
@@ -18,3 +18,16 @@ export const updateRouterQuery = (
     shallow: true,
   });
 };
+
+export const setURLParam = (param: string, value: string) => {
+  const qs = new URL(window.location.href).searchParams;
+  qs.delete(param);
+  qs.append(param, value);
+  const newUrl =
+    window.location.protocol +
+    "//" +
+    window.location.host +
+    window.location.pathname +
+    `?${qs.toString()}`;
+  window.history.replaceState({ path: newUrl }, "", newUrl);
+};
```

Comment:
I really like how much simpler things became because of this 👏 

Nit: I think it should be enough to just do `?${qs}`, as when it's interpolated by default it looks for .toString() method

Reply:
Done, and used string interpolation instead of the + ;)